tests/capture/test_capture_transforms.py

```diff
@@ -0,0 +1,432 @@
+        inval = trace.pure(1.5)
+        interpreter = TransformInterpreter(program)
+        interpreter._trace = trace
+
+        outval = interpreter.read_with_trace(inval)
+        assert outval is inval
+
+    @pytest.mark.parametrize(
+        "inval",
+        [
+            1.5,
+            0,
+            1 + 0j,
+            jnp.array(1.0),
+            # Creates a TransformTracer, but its _trace does not belong to the interpreter
+            init_test_variables()[2].pure(-1),
+            # Creates a nested TransformTracer, but its _trace does not belong to the interpreter
+            init_test_variables()[2].pure(init_test_variables()[2].pure(-2)),
+        ],
+    )
+    def test_read_with_trace_unboxed(self, inval):
+        """Test that values that are not boxed into tracers belonging to the interpreter
+        are boxed in TransformTracers."""
+        program, _, trace = init_test_variables()
+        interpreter = TransformInterpreter(program)
+        interpreter._trace = trace
+
+        outval = interpreter.read_with_trace(inval)
+        assert outval == trace.pure(inval)
+
+    def test_interpret_operation_eqn_dropped(self):
+        """Test that operator equations that output a DropVar are interpreted correctly."""
+
+    def test_interpret_operation_eqn_not_dropped(self):
+        """Test that operator equations that do not output a DropVar are not transformed."""
+
+    def test_interpret_operation_eqn_operator_input(self):
+        """Test that operator equations that include another operator as input are
+        interpreted correctly."""
+
+    def test_interpret_measurement_eqn(self):
+        """Test that measurement equations are interpreted correctly."""
+
+    def test_eval_adjoint_error(self):
+        """Test that an error is raised when trying to interpret Adjoint primitives."""
+
+        def adj_func(x):
+            qml.RX(x, 0)
+
+        def f(x):
+            qml.adjoint(adj_func)(x)
+            return qml.expval(qml.Z(0))
+
+        args = (1.5,)
+        jaxpr = jax.make_jaxpr(f)(*args)
+        program, _, _ = init_test_variables(transforms=[change_rotations])
+        interpreter = TransformInterpreter(program)
+
+        with pytest.raises(NotImplementedError):
+            _ = interpreter.eval(jaxpr.jaxpr, jaxpr.consts, *args)
+
+    def test_eval_ctrl_error(self):
+        """Test that an error is raised when trying to interpret Control primitives."""
+
+        def ctrl_func(x):
+            qml.RX(x, 0)
+
+        def f(x):
+            qml.ctrl(ctrl_func, [1, 2])(x)
+            return qml.expval(qml.Z(0))
+
+        args = (1.5,)
+        jaxpr = jax.make_jaxpr(f)(*args)
+        program, _, _ = init_test_variables(transforms=[change_rotations])
+        interpreter = TransformInterpreter(program)
+
+        with pytest.raises(NotImplementedError):
+            _ = interpreter.eval(jaxpr.jaxpr, jaxpr.consts, *args)
+
+
+@pytest.mark.integration
+class TestTransformInterpreterIntegration:
```

Comment:
I guess the intention is to implement these tests soon (similar for other in the same file)